c2cgeoportal/lib/__init__.py

```diff
@@ -152,8 +152,11 @@ def __call__(self, request, elements, kw):
             from c2cgeoportal.lib.cacheversion import get_cache_version
             query["cache_version"] = get_cache_version()
 
-        if self.role and request.user and request.user.role:
-            query["role"] = request.user.role.id
+        if self.role and request.user:
```

Comment:
I don't think we should remove it. In some cases (one, actually), we don't want to invalidate the cache if the user changes identity. This one:
```python
    config.add_route(
        "layers_enumerate_attribute_values",
        "/layers/{layer_name}/values/{field_name}",
        request_method="GET",
        pregenerator=C2CPregenerator(),
    )
```